### DISCUSSION TITLE
How does the CLI automatically create SHA256-hash of a document?

### DISCUSSION THREAD

**Discussion by dnlsandiego** [0no-co/gql.tada#246] [Q&A] [answered]
Hi folks, quick clarification question on `graphql.persisted()` and the `gql-tada generate-persisted` command as I might be missing something on getting hashes to be automatically generated. Specifically on this part on the docs:

> @0no-co/graphqlsp will then check that the document is available and offer a code action to automatically update the hash to a SHA256-hash of the document.

What does "offer a code action" mean here? Is this a prompt that the CLI should show when running `gql-tada generate-persisted`? I'm calling `const persistedOperation = graphql.persisted<typeof query>('sha256:x');` then running the command, the JSON file gets generated but I don't see the hash of the document string as the key. What step am I missing here to get the CLI to automatically hash the document string?

**Comment by JoviDeCroock** [comment, marked as answer]
The LSP will show a prompt you with a code action, the CLI only extracts hashes so it does not generate anything at the moment

**Reply by dnlsandiego** [reply]
And by code action, is it the message `The persisted document's hash is outdated`? 

Actually, I think I'm still not quite following, mostly because when reading through [generated-ids](https://gql-tada.0no.co/guides/persisted-documents#using-generated-ids), my understanding on this section is that having a call:

```ts
// file: pokemon-query.ts
const persistedQuery = graphql.persisted<typeof pokemonsQuery>(
  "POKEMONS_LIST_ID"
);
```
Then running 
```bash
gql-tada generate-persisted --output persisted.json
```
The `"POKEMONS_LIST_ID"` string in the file `pokemon-query.ts` gets replaced with `"sha256:89e47d4f32b4ff76296844ff260d2878bf1829d30706fc7fc92de0fc66c2a4cf"` then in the `persisted.json` file,

```json
{
  "sha256:89e47d4f32b4ff76296844ff260d2878bf1829d30706fc7fc92de0fc66c2a4cf": {....}
}
```

Is this the expected outcome for this flow?

**Reply by kitten** [reply]
No, the two sections in the docs are unrelated — as it says in that particular section:
> Each entry is keyed by a document ID and has a value of the GraphQL detected document.

So, the keys of the JSON will be whichever string literal is used as the first argument to `graphql.persisted`
